src/main/java/joe/task/TaskList.java

```diff
@@ -0,0 +1,114 @@
+package joe.task;
+
+import java.util.ArrayList;
+
+/**
+ * The {@code TaskList} class represents a collection of tasks and provides methods
+ * to interact with this collection, such as adding, removing, marking, unmarking,
+ * and finding tasks.
+ */
+public class TaskList {
+    /** The list that contains all the tasks. */
+    private final ArrayList<Task> taskList;
+
+    /**
+     * Constructs an empty {@code TaskList}.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a {@code TaskList} with an initial list of tasks.
+     *
+     * @param tasks An {@code ArrayList} of tasks to initialize the list.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.taskList = new ArrayList<>();
+        this.taskList.addAll(tasks);
+    }
+
+    /**
+     * Converts the {@code TaskList} to an {@code ArrayList}.
+     *
+     * @return The {@code ArrayList} containing all the tasks.
+     */
+    public ArrayList<Task> toArrayList() {
+        return this.taskList;
+    }
```

Comment:
Would a getter method fit better here since your are returning this.taskList directly? Perhaps getArrayList()?